core-plugins/src/test/java/co/cask/hydrator/plugin/batch/source/FileBatchSourceTest.java

```diff
@@ -146,4 +148,47 @@ public void testNotPresentFolder() throws Exception {
     workflowManager.waitForFinish(5, TimeUnit.MINUTES);
     Assert.assertEquals("FAILED", workflowManager.getHistory().get(0).getStatus().name());
   }
+
+  @Test
+  public void testRecursiveFolders() throws Exception {
+    Map<String, String> sourceProperties = new ImmutableMap.Builder<String, String>()
+      .put(Constants.Reference.REFERENCE_NAME, "TestCase")
+      .put(Properties.File.PATH, "src/test/resources/")
+      .put(Properties.File.FILE_REGEX, "fileBatchSource.*")
```

Comment:
I don't think this test case is complete. I want to be able to do something like this: My base directory is /tmp/data, it contains a/x/some.txt, a/y/other.txt, b/x/this.txt, b/y/that.txt, and I want to read recursively and accept all files that match */x/*.txt. Is that possible?